src/esp/gfx/PTexMeshShader.cpp

```diff
@@ -87,5 +90,38 @@ PTexMeshShader& PTexMeshShader::bindAdjFacesBufferTexture(
   return *this;
 }
 
+PTexMeshShader& PTexMeshShader::setMVPMatrix(const Magnum::Matrix4& matrix) {
+  setUniform(MVPMatrixUniform_, matrix);
+  return *this;
+}
+
+PTexMeshShader& PTexMeshShader::setExposure(float exposure) {
+  setUniform(exposureUniform_, exposure);
+  return *this;
+}
+PTexMeshShader& PTexMeshShader::setGamma(float gamma) {
+  // Careful: we set its inverse, not gamma directly
+  setUniform(gammaInverseUniform_, 1.0f / gamma);
+  return *this;
+}
+
+PTexMeshShader& PTexMeshShader::setSaturation(float saturation) {
+  setUniform(saturationUniform_, saturation);
+  return *this;
+}
+
+PTexMeshShader& PTexMeshShader::setAtlasTextureSize(
+    Magnum::GL::Texture2D& texture,
+    uint32_t tileSize) {
+  setUniform(tileSizeUniform_, (int)tileSize);
+
+  // Image size in given mip level 0
+  int mipLevel = 0;
+  int widthEntry = 0;
+  const auto width = texture.imageSize(mipLevel)[widthEntry];
```

Comment:
I get that this code was already there and you just moved it, but what about `texture.imageSize(mipLevel).x()`?

Reply:
Good suggestion.